### PR TITLE
Refactor some duplicated methods in *action classes #6448

### LINKED ISSUE #6448: Refactor some duplicated methods in InstructorFeedback*Action classes
InstructorFeedbackAddAction and InstructorFeedbackEditSaveAction have implementations of extractFeedbackSessionData() which are almost completely identical with small difference (e.g, error handling, setting 1-2 fields differently).

Similarly for extractFeedbackQuestionData(String) for InstructorFeedbackQuestionAddAction and InstructorFeedbackQuestionVisibilityMessageAction.

### PR DESCRIPTION
**Pull request by AnaghHegde**
Fixes #6448 

**Outline of Solution**

 Continued from #6480.

### CODE REVIEW

**Comment by AnaghHegde** [comment]
@samsontmr Neglect the first 2 commits. I was working on the other issue. By mistake i committed and i have changed it to original in the commit 31d23c2.

File: src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        return courses;
+    }
+
+    /**
+     * Gets a Map with courseId as key, and InstructorAttributes as value.
+     */
+    protected Map<String, InstructorAttributes> loadCourseInstructorMap(boolean omitArchived) {
+        Map<String, InstructorAttributes> courseInstructorMap = new HashMap<String, InstructorAttributes>();
+        List<InstructorAttributes> instructors = logic.getInstructorsForGoogleId(account.googleId,
+                                                                                 omitArchived);
```
Comment (samsontmr) [inline src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java]: We've decided to move away from custom indentation. Use 8 spaces but insert the line break at as high level as possible.

**Comment by samsontmr** [comment]
Refer to [this PR](https://github.com/TEAMMATES/teammates/pull/6578#issuecomment-284149244) if you are still unclear on how to organize your commits. Use canihasreview as well.

**Comment by AnaghHegde** [comment]
Should i need to change the indentation in all the places meaning : the part of the code which i did not change during this PR ?.

**Comment by samsontmr** [comment]
> Should i need to change the indentation in all the places meaning : the part of the code which i did not change during this PR ?.

You may change it in the classes you modify, but make sure it follows the one commit per logical change rule--meaning that there should be one (and ideally only one) commit with only indentation changes.

**Comment by AnaghHegde** [comment]
I apologize for the inconvenience caused, all i did was `git push - u origin branch-name:remote-branch-name`. But it somehow got deleted.

**Comment by AnaghHegde** [comment]
Ready for review after the commit 06fb6da .

**Comment by AnaghHegde** [comment]
Ready for review.

**Comment by samsontmr** [comment]
Do you know how to squash and rebase? A [clean commit history](https://www.notion.so/Keeping-Commit-Histories-Clean-0f717c4e802c4a0ebd852cf9337ce5d2) means that I should not be seeing commits like `fix PMD error`, `revert changes to commit 26443c1`, `fix checkstyle issues` etc. All these `fix *` actions should be committed together with the modification you made that caused the issue. Finally, you should ideally be using `rebase` to update your branch instead of `merge`.

**Comment by AnaghHegde** [comment]
Yes i know squashing of commits. what all commits needs to squashed ?. I guess last 5 or 6 commits ?.

**Comment by AnaghHegde** [comment]
I think this is what you meant. Next time i will follow these steps. Ready for review.

**Comment by samsontmr** [comment]
> I think this is what you meant. Next time i will follow these steps. Ready for review.

Yes, do this for your earlier commits as well (see my previous comment).

**Comment by AnaghHegde** [comment]
I assume from this c50f734 commit ?

**Comment by samsontmr** [comment]
From the beginning.

**Comment by AnaghHegde** [comment]
All the commits ?. what should be the commit message for this ?

**Comment by samsontmr** [comment]
As I said in a previous comment, do not squash everything into a single commit.
> I should not be seeing commits like `fix PMD error`, `revert changes to commit 26443c1`, `fix checkstyle issues` etc. All these `fix *` actions should be committed together with the modification you made that caused the issue.

**Comment by samsontmr** [comment]
Also see [here](https://www.notion.so/Keeping-Commit-Histories-Clean-0f717c4e802c4a0ebd852cf9337ce5d2) for an example of a clean vs unclean history.

**Comment by AnaghHegde** [comment]
I don't know what is going wrong every time i push the branch is getting deleted. I will clean my master repo and then i will open a new PR. Sorry for the inconvenience.

**Pull request closed without merging**